File: Leetcode/Review/20211004/firstUniqChar.go
package _0211004

func firstUniqChar(s string) byte {
	hash:=map[byte]int{}
	for i:=0;i<len(s);i++{
		hash[s[i]]++
	}
	for i:=0;i<len(s);i++{
		if hash[s[i]]==1{
			return s[i]
		}
	}
	return ' '
}

File: Leetcode/剑指offer/easy/20210914/firstUniqChar.cpp
class Solution {
public:
    char firstUniqChar(string s) {
    int c[26] = {0};
	int size = s.size();
	if (!size)return ' ';
	for (int i = 0; i < size; i++){c[s[i] - 'a']++;}
	for (int i = 0; i < size; i++)
	{
		if (c[s[i]-'a'] == 1)return s[i];
	}
	return ' ';
    }
};

